Source/Private/Invoke-ExecCommandCore.ps1

```diff
@@ -0,0 +1,74 @@
+#   Borrowed from our friendly provider: https://github.com/dotnet/roslyn/blob/master/src/Setup/Installer/tools/utils.ps1
+
+function Invoke-ExecCommandCore {
+    param (
+        [Parameter(mandatory = $true)]
+        [string] $Command,
+        [string] $CommandArgs,
+        [switch] $PassThru = $true
+    )
+
+    $result = [CommandCoreResult]::new()
+    $result.Command = $Command
+    $result.CommandArgs = $CommandArgs
+
+    $startInfo = New-Object System.Diagnostics.ProcessStartInfo
+    $startInfo.FileName = $Command
+    $startInfo.Arguments = $CommandArgs
+
+    $startInfo.UseShellExecute = $false
+    $startInfo.WorkingDirectory = Get-Location
+
+    #   Always redirect output
+    $startInfo.RedirectStandardOutput = $true
+    $startInfo.RedirectStandardError = $true
+    $startInfo.CreateNoWindow = $true
```

Comment:
Consider extracting to a method, this function has too much responsibility.

Reply:
In progress